test/test_dataflow.py

```diff
@@ -1,51 +1,78 @@
 from unittest import TestCase
+from unittest.mock import patch
+
 import apache_beam as beam
 from apache_beam import Pipeline
 from apache_beam.options.pipeline_options import PipelineOptions
-from apache_beam.testing.util import assert_that, equal_to
-
+from apache_beam.runners.portability.fn_api_runner.fn_runner import RunnerResult
+from apache_beam.testing.test_pipeline import TestPipeline
+from collections import defaultdict
 from bigflow import JobContext, Workflow
 
 from bigflow.dataflow import BeamJob
 
 
 class CountWordsFn(beam.DoFn):
+    def __init__(self, save):
+        super().__init__()
+        self.save = save
+
     def process(self, element, *args, **kwargs):
         word, count = element
+        self.save.counter[word] += len(count)
         yield word, len(count)
 
 
+class Save(object):
+    counter = defaultdict(int)
+
+
+class Saver(beam.PTransform):
+    def __init__(self, save):
+        super().__init__()
+        self.save = save
+
+    def expand(self, records_to_delete):
+        return records_to_delete \
+               | "SaveCountedWords" >> beam.ParDo(
+            CountWordsFn(self.save))
+
+
 class CountWordsDriver:
-    def __init__(self):
+    def __init__(self, saver):
         self.result = None
         self.context = None
         self.pipeline = None
+        self.saver = saver
 
     def run(self, pipeline: Pipeline, context: JobContext, driver_arguments: dict):
         words_input = pipeline | 'LoadingWordsInput' >> beam.Create(driver_arguments['words_to_count'])
-        self.result = (words_input
-                       | 'FilterWords' >> beam.Filter(lambda w: w in driver_arguments['words_to_filter'])
-                       | 'MapToCount' >> beam.Map(lambda w: (w, 1))
-                       | 'GroupWords' >> beam.GroupByKey()
-                       | 'CountWords' >> beam.ParDo(CountWordsFn()))
+        words_input | 'FilterWords' >> beam.Filter(lambda w: w in driver_arguments['words_to_filter']) \
+        | 'MapToCount' >> beam.Map(lambda w: (w, 1))\
+        | 'GroupWords' >> beam.GroupByKey()\
+        | 'CountWords' >> self.saver
```

Comment:
nit: AFAIK it is not recommended to use '\' line continuation. https://www.python.org/dev/peps/pep-0008/#maximum-line-length

Reply:
This is how we do it in our scripts

Reply:
'beam' official documentation [uses](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/wordcount_minimal.py#L112) different style (with extra `()` but without `\`).
Maybe we should consider using the "official" approach ?

Reply:
Changed